### PR TITLE
refactor OutGoingOperationsPool

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- added `OutGoingOperationsPool` in RunType components
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/runType/runTypeComponents.go [hunk 1]
```diff
@@ -156,6 +158,8 @@ func (rcf *runTypeComponentsFactory) Create() (*runTypeComponents, error) {
 		return nil, fmt.Errorf("runTypeComponentsFactory - NewAccountCreator failed: %w", err)
 	}
 
+	outGoingOperationsPool := disabled.NewDisabledOutGoingOperationPool()
```
Comment (mariusmihaic) [inline factory/runType/runTypeComponents.go:161]: You can add it directly in returned obj

File: factory/runType/sovereignRunTypeComponents.go [hunk 2]
```diff
@@ -157,6 +159,9 @@ func (rcf *sovereignRunTypeComponentsFactory) Create() (*runTypeComponents, erro
 		return nil, fmt.Errorf("sovereignRunTypeComponentsFactory - NewSovereignAccountCreator failed: %w", err)
 	}
 
+	expiryTime := time.Second * time.Duration(rcf.cfg.OutgoingSubscribedEvents.TimeToWaitForUnconfirmedOutGoingOperationInSeconds)
+	outGoingOperationsPoolCreator := sovereignFactory.NewOutGoingOperationPool(expiryTime)
```
Comment (mariusmihaic) [inline factory/runType/sovereignRunTypeComponents.go:163]: This is not a creator, it is the outgoing op pool
```suggestion
	outGoingOperationsPool:= sovereignFactory.NewOutGoingOperationPool(expiryTime)
```

File: process/block/baseProcess_test.go [hunk 3]
```diff
@@ -813,15 +813,23 @@ func TestCheckProcessorNilParameters(t *testing.T) {
 		{
 			args: func() blproc.ArgBaseProcessor {
 				args := createArgBaseProcessor(coreComponents, dataComponents, bootstrapComponents, statusComponents)
-				args.RunTypeComponents = &mock.RunTypeComponentsStub{AccountCreator: &stateMock.AccountsFactoryStub{}, DataCodec: nil}
+				args.RunTypeComponents = &mock.RunTypeComponentsStub{AccountCreator: &stateMock.AccountsFactoryStub{}, OutGoingOperationsPool: nil}
```
Comment (mariusmihaic) [inline process/block/baseProcess_test.go:816]: Same remark as in previous PR

File: process/block/interface.go [hunk 4]
```diff
@@ -45,16 +45,6 @@ type crossNotarizer interface {
 	getLastCrossNotarizedHeaders() []bootstrapStorage.BootstrapHeaderInfo
 }
 
-// OutGoingOperationsPool defines the behavior of a timed cache for outgoing operations
-type OutGoingOperationsPool interface {
```
Comment (mariusmihaic) [inline process/block/interface.go:49]: It was better to keep the interface definition at user/caller side, not where the implementation of the interface is.
Can stay as it is, though....

File: process/mock/runTypeComponentsStub.go [hunk 5]
```diff
@@ -28,6 +31,10 @@ func (r *RunTypeComponentsStub) AccountsCreator() state.AccountFactory {
 	return r.AccountCreator
 }
 
+func (r *RunTypeComponentsStub) OutGoingOperationsPoolHandler() sovereignBlock.OutGoingOperationsPool {
```
Comment (mariusmihaic) [inline process/mock/runTypeComponentsStub.go:34]: Add missing comm

File: consensus/spos/bls/sovereignSubRoundEnd.go [hunk 6]
```diff
@@ -5,24 +5,25 @@ import (
 	"encoding/hex"
 	"fmt"
 
+	"github.com/multiversx/mx-chain-go/consensus/spos"
+	sovereignBlock "github.com/multiversx/mx-chain-go/dataRetriever/dataPool/sovereign"
```
[the remark this answers is not in the document]
Comment (axenteoctavian) [reply consensus/spos/bls/sovereignSubRoundEnd.go:9]: I created an interface

File: factory/runType/sovereignRunTypeComponents.go [hunk 7]
```diff
@@ -161,11 +161,7 @@ func (rcf *sovereignRunTypeComponentsFactory) Create() (*runTypeComponents, erro
 	}
 
 	expiryTime := time.Second * time.Duration(rcf.cfg.OutgoingSubscribedEvents.TimeToWaitForUnconfirmedOutGoingOperationInSeconds)
-	outGoingOperationsPoolCreator := sovereignFactory.NewOutGoingOperationPool(expiryTime)
-
-	dataCodec := rcf.dataCodec
-
-	topicsChecker := rcf.topicsChecker
+	outGoingOperationsPool := sovereignFactory.NewOutGoingOperationPool(expiryTime)
```
Comment (mariusmihaic) [inline factory/runType/sovereignRunTypeComponents.go:164]: Put directly in return

**Review by raduchis — APPROVED** [review]

File: process/mock/runTypeComponentsStub.go [hunk 8]
```diff
@@ -36,8 +36,8 @@ func (r *RunTypeComponentsStub) OutGoingOperationsPoolHandler() sovereignBlock.O
 	return r.OutGoingOperationsPool
 }
 
-// DataCodecHandler -
-func (r *RunTypeComponentsStub) DataCodecHandler() sovereign.DataDecoderHandler {
+// DataDecoderHandler -
```
Comment (mariusmihaic) [inline process/mock/runTypeComponentsStub.go:39]:
```suggestion
// DataCodecHandler -
```

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

**Review by raduchis — DISMISSED** [review]

**Review by mariusmihaic — DISMISSED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**